app/assets/javascripts/modules/inline-image-modal.js

```diff
@@ -0,0 +1,79 @@
+/* global MultiSectionViewer */
+
+function InlineImageModal (trigger) {
+  this.$trigger = trigger
+  this.$modal = document.getElementById('modal')
+}
+
+InlineImageModal.prototype.init = function () {
+  var $module = this
+
+  if (!this.$trigger || !this.$modal) {
+    return
+  }
+
+  this.$multiSectionViewer = new MultiSectionViewer(this.$modal.querySelector('[data-module="multi-section-viewer"]'))
+  this.$multiSectionViewer.init()
+
+  this.$trigger.addEventListener('click', function (event) {
+    event.preventDefault()
+    $module.performAction($module.$trigger)
+  })
+}
+
+InlineImageModal.prototype.fetchModalContent = function (url) {
+  var controller = new window.AbortController()
+  var headers = { 'X-Requested-With': 'XMLHttpRequest' }
+  var options = { credentials: 'include', signal: controller.signal, headers: headers }
+  setTimeout(function () { controller.abort() }, 5000)
```

Comment:
Is there intentions of consistency with the prefixing of window to things? I notice this isn't prefixed with window whereas fetch and promise are, and then we've also not prefixed MultiSectionViewer

Reply:
Good point - I was going from the `MarkdownEditor` component, which does the same thing for a referenced class using `global`, but both are unnecessary by using `window`. The linter doesn't actually complain about `setTimeout`, but does for a bunch of other things, which is why they have the `window.` prefix. Unless it's causing a problem (including the linter complaining about it), I'm inclined to not add the prefix.